src/components/DatePickerRange.react.js

```diff
@@ -0,0 +1,298 @@
+import React, {Component, PropTypes} from 'react';
+import { DateRangePicker } from 'react-dates';
+import moment from 'moment';
+
+/**
+ * DatePickerRange is a tailor made component designed for selecting
+ * timespan across multiple days off of a calendar.
+ *
+ * The DatePicker integrates well with the Python datetime module with the
+ * startDate and endDate being returned in a string format suitable for
+ * creating datetime objects.
+ */
+export default class DatePickerRange extends Component {
+    constructor(props) {
+      super(props);
+      this.props.startDate = this.props.start_date;
+      this.props.endDate = this.props.end_date;
+      this.props.initialVisibleMonth = this.props.initial_visible_month;
+      this.props.minDateRange = this.props.min_date_range;
+      this.props.maxDateRange = this.props.max_date_range;
+      const momentProps = this.convertPropsToMoment(props);
+      this.state = {
+        focusedInput: props.autoFocusEndDate ? momentProps.startDate : momentProps.endDate,
+        startDate: momentProps.startDate,
+        endDate: momentProps.endDate,
+        initialVisibleMonth: momentProps.initialVisibleMonth,
+        minDateRange: momentProps.min,
+        maxDateRange: momentProps.max,
+        initialStartDate: this.props.start_date,
+        initialEndDate: this.props.end_date,
+        firstInitialVisibleMonth: this.props.initial_visible_month,
+        initialMinDateRange: this.props.min_date_range,
+        initialMaxDateRange: this.props.max_date_range
+      };
+
+    }
+
+    convertPropsToMoment(props) {
+      let startDate, endDate=null; let initialVisibleMonth = moment(props.startDate);
+      if(props.startDate != undefined) { startDate = moment(props.startDate); }
+      if(props.endDate != undefined) { endDate = moment(props.endDate); }
+      if(props.initialVisibleMonth != undefined) { initialVisibleMonth = moment(props.initialVisibleMonth); }
+      let min, max;
+      if(props.minDateRange != undefined && props.maxDateRange != undefined) {
+        min = moment(props.minDateRange);
+        max = moment(props.maxDateRange);
+      }
+      try{
+        if(startDate.isAfter(endDate)){
+          endDate = null;
+        }
+      } catch (TypeError) {
+        // Continue regardless of error
+      }
+
+      return {startDate, endDate, initialVisibleMonth, min, max}
+    }
+
+    componentWillReceiveProps(newProps) {
+      newProps.startDate = newProps.start_date;
```

Comment:
we shouldn't mutate `newProps` - this will modify the props for the parent component which is unpredicable. Instead, you should make a new object and transfer the properties over to it.